packages/client/src/hooks/useCheckOauthCode.ts

```diff
@@ -0,0 +1,18 @@
+import { fetchOAuthStepThreeGetApproveFromApiPracticum } from 'src/store/auth/AuthActions';
+import { useBoundAction } from './useBoundAction';
+import { useMountEffectOneCall } from './useMountEffectOneCall';
+
+export const useCheckOauthCode = () => {
+  const handler = useBoundAction(code =>
+    fetchOAuthStepThreeGetApproveFromApiPracticum(code)
+  );
+
+  useMountEffectOneCall(() => {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const code = urlParams.get('code');
+    if (code) {
+      handler(code);
```

Comment:
```suggestion
      handler(code);
            urlParams.delete('code');
      window.history.replaceState({}, '', env.REDIRECT_URI);
```

после успешной авторизации убираем code из адресной строки